src/plugins/WebOfTrust/util/SynchronousDelayedBackgroundJob.java

```diff
@@ -0,0 +1,231 @@
+        this.job = job;
+        this.name = name;
+        this.defaultDelay = delay;
+    }
+
+    /**
+     * Triggers the execution of the job after either aggregating triggers for at most the default
+     * delay, or as soon as the currently running job finishes, whichever comes last.
+     * If this background job was terminating or has terminated prior to the invocation of this
+     * method, this method has no effect and returns immediately.
+     * This method waits for the first job to start after the trigger to finish before returning,
+     * except when this method is called from the background job itself, in that case this method
+     * does not wait for the job to finish (otherwise a deadlock would be the result).
+     * If the calling thread is interrupted while waiting for the job to finish execution, this
+     * method will return as soon as possible, without waiting for the job to finish, and sets the
+     * thread interruption flag.
+     */
+    @Override
+    public synchronized void triggerExecution() {
+        triggerExecution(defaultDelay);
+    }
+
+    /**
+     * Triggers the execution of the job after either aggregating triggers for at most the given
+     * delay, or as soon as the currently running job finishes, whichever comes last. Negative
+     * delays are treated as to zero delays.
+     * If this background job was terminating or has terminated prior to the invocation of this
+     * method, this method has no effect and returns immediately.
+     * This method waits for the first job to start after the trigger to finish before returning,
+     * except when this method is called from the background job itself, in that case this method
+     * does not wait for the job to finish (otherwise a deadlock would be the result).
+     * If the calling thread is interrupted while waiting for the job to finish execution, this
+     * method will return as soon as possible, without waiting for the job to finish, and sets the
+     * thread interruption flag.
+     * @param delay the maximum trigger aggregation delay in milliseconds
+     */
+    @Override
+    public synchronized void triggerExecution(long delay) {
+        if (isTerminating) {
+            return;
+        }
+        triggerExecutionAsynchronously(delay);
+        // When called from the job thread, don't wait for the job to finish: this would cause a
+        // deadlock.
+        if (runningJobThread == Thread.currentThread()) {
+            return;
+        }
+        // Wait for the relevant execution to complete.
+        long waitForRunCount;
+        if (runningJobThread == null) {
+            // Wait for next run to finish
+            waitForRunCount = runCount + 1;
+        } else {
+            // Wait for next run after current run to finish
+            waitForRunCount = runCount + 2;
+        }
+        while (runCount < waitForRunCount) {
+            try {
+                wait();
```

Comment:
> Using a Ticker to replace a few lines of code here sounds rather pointless. This entire class is less than 250 LOC, most of which are comments.

Heh, what would you say if I told me that it took me hours to review the class?
Threading is a complex thing for the human mind, and 250 LOC of threading can take hours to understand :| The difficulty is also greatly amplified by not having writing such things yourself - I didn't know that this can take so long before I reviewed this one, I was surprised myself because I have written larger, more complex threaded code.
Eliminating the whole threading logic by recycling the TickerDelayedBackGroundJob to power the class would prevent possible future review havoc.
I wouldn't say its a blocker though, you already wrote it, I reviewed it, so it can stay as is if you want it to stay. I'd say only re-invent it if the shutdown issue fixes which are needed would be less work for you that way.